src/thirty_two.rs

```diff
@@ -215,15 +227,59 @@ impl Hasher for XxHash {
     }
 }
 
+impl Digest for XxHash {
+    type OutputSize = U8;
```

Comment:
This is a *bit* of a lie, as the point of the 32-bit hash is that it’s actually only… 32bits! However, we cast it to `u64` to fit the `Hash` trait. We *should* move the actual hashing logic out of the `Hash` trait and into the type itself, returning the correct size value. Then do just the cast in the `Hash` trait` implementation.

However, for this PR, you don’t need to worry about that. I’d just truncate back to a `u32` for now. Later on, we can cancel out the double casts.

Reply:
Fixing now :)